src/parse.cpp

```diff
@@ -216,3 +216,127 @@ template <> double chemfiles::parse(string_view input) {
     // Return signed and scaled floating point result.
     return sign * (frac ? (value / scale) : (value * scale));
 }
+
+static const auto digits_upper = std::string("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+static const auto digits_lower = std::string("0123456789abcdefghijklmnopqrstuvwxyz");
+
+static uint8_t digit_to_value(char c) {
+    if (c >= '0' && c <= '9') {
+        return c - '0';
+    }
+
+    if (c >= 'A' && c <= 'Z') {
+        return c - 'A' + 10;
+    }
+
+    // We know all the places this function is used, so this should be fine
+    return c - 'a' + 10;
+}
+
+static std::string encode_pure(const std::string& digits, uint64_t value) {
+    if (value == 0) {
+        return std::string(digits, 1);
+    }
+    
+    auto n = digits.length();
+    std::string result;
+    while (value != 0) {
+        auto rest = value / n;
+        result += digits[value - rest * n];
+        value = rest;
+    }
+
+    std::reverse(result.begin(), result.end());
+    return result;
+}
+
+static uint64_t decode_pure(string_view s) {
+    uint64_t result = 0;
+    auto n = digits_upper.length();
+    for (auto c : s) {
+        result *= n;
+        result += digit_to_value(c);
+    }
+    return result;
+}
+
+static int64_t pow_int(uint64_t base, uint64_t power) {
```

Comment:
Not sure I understand the difference between this function and just `std::pow`, could you add a short comment?